uaclient/log.py

```diff
@@ -58,3 +63,87 @@ def format(self, record: logging.LogRecord) -> str:
 
         local_log_record["extra"] = extra_message_dict
         return json.dumps(list(local_log_record.values()))
+
+
+class LogFile(enum.Enum):
+    MAIN = 0
+    TIMER = 1
+    DAEMON = 2
+
+    def default_file(self) -> str:
+        if self == self.MAIN:
+            return CONFIG_DEFAULTS["log_file"]
+        raise NotImplementedError("Implement me for {}".format(self))
+
+    def extract_log_file(self, ua_cfg: config.UAConfig) -> str:
+        if self == self.MAIN:
+            return ua_cfg.log_file
+        raise NotImplementedError("Implement me for {}".format(self))
+
+
+def setup_logging(console_level, log_level, log_file=None, logger=None):
```

Comment:
Being this used on more than the cli entry-point, I thought it was a good idea to move it here.